openupgrade_scripts/scripts/fleet/18.0.0.1/post-migration.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2025 Open Net Sàrl
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+
+
+def _set_power_unit(env):
+    # if horspower is defined, set power_unit to horsepower
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE fleet_vehicle_model
+        SET power_unit='horsepower'
+        WHERE horsepower IS NOT NULL
```

Comment:
Since the field is numeric, it does not contain NULL values. With your code, the column `power_unit` is set to `horsepower` in all rows, including rows where it should not apply (see the image).
```suggestion
        WHERE horsepower > 0
```
<img width="1481" height="469" alt="image" src="https://github.com/user-attachments/assets/7afda61d-5035-4d3c-9a06-78e45f4516c2" />

Reply:
It worked when I tested it but you are right in odoo once a value is applied to an integer you cannot go back to NULL in database even if you set it to 0 in Odoo. Integer fields are NULL when they have never been assigned a value so > 0 is indeed better.
<img width="234" height="300" alt="odoo_int" src="https://github.com/user-attachments/assets/3ce63561-727c-4d4b-be41-e7e19bea7de2" />